support/scheduler-client/client.go

```diff
@@ -0,0 +1,98 @@
+/*******************************************************************************
+ * Copyright 2017 Dell Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ * @microservice: support-scheduler-client-go library
+ * @author: Vino Yang, Tencent
+ * @version: 0.5.0
+ *******************************************************************************/
+
+package scheduler
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/edgexfoundry/edgex-go/core/domain/models"
+	"io"
+	"net/http"
+)
+
+// Common http const
+const (
+	HttpPostMethod     = "POST"
+	ContentType        = "Content-Type"
+	ContentTypeJsonVal = "application/json"
+)
+
+// Struct to represent the scheduler client
+type SchedulerClient struct {
+	RemoteScheduleUrl      string
+	RemoteScheduleEventUrl string
+	RemoteCallbackAlertUrl string
+	OwningService          string
+}
+
+// Function to send a schedule to the remote scheduler server
+func (schedulerClient SchedulerClient) SendSchedule(schedule models.Schedule) error {
+	client := &http.Client{}
+
+	// get the json request body
+	requestBody, err := schedule.MarshalJSON()
+	if err != nil {
+		fmt.Println(err.Error())
+		return err
+	}
+
+	return doPost(schedulerClient.RemoteScheduleUrl, bytes.NewBuffer(requestBody), client)
+}
+
+// Function to send a schedule event to the remote scheduler server
+func (schedulerClient SchedulerClient) SendScheduleEvent(scheduleEvent models.ScheduleEvent) error {
+	client := &http.Client{}
+
+	// get the json request body
+	requestBody, err := scheduleEvent.MarshalJSON()
+	if err != nil {
+		fmt.Println(err.Error())
+		return err
+	}
+
+	return doPost(schedulerClient.RemoteScheduleEventUrl, bytes.NewBuffer(requestBody), client)
+}
+
+// Function to do post request
+func doPost(url string, binaryReqBody io.Reader, client *http.Client) error {
+	// create request
+	req, err := http.NewRequest(HttpPostMethod, url, binaryReqBody)
+	req.Header.Add(ContentType, ContentTypeJsonVal)
+
+	if err != nil {
+		fmt.Println(err.Error())
+		return err
+	}
+
+	// send request call
+	sendRequest(client, req)
+
+	return nil
+}
+
+// Function to actually make the HTTP request
+func sendRequest(client *http.Client, req *http.Request) {
+	resp, err := client.Do(req)
+	if err == nil {
+		defer resp.Body.Close()
+		resp.Close = true
+	} else {
+		fmt.Println(err.Error())
```

Comment:
No printing on the command line - use logs.

Reply:
@drasko thanks for reviewing. Does you mean use edgex's `LoggingClient`?